samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/ChatHistoryController.cs

```diff
@@ -28,54 +28,62 @@ public class ChatHistoryController : ControllerBase
     private readonly ILogger<ChatHistoryController> _logger;
     private readonly ChatSessionRepository _sessionRepository;
     private readonly ChatMessageRepository _messageRepository;
-    private readonly PromptsOptions _promptOptions;
+    private readonly ChatParticipantRepository _participantRepository;
     private readonly ChatMemorySourceRepository _sourceRepository;
+    private readonly PromptsOptions _promptOptions;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="ChatHistoryController"/> class.
     /// </summary>
     /// <param name="logger">The logger.</param>
     /// <param name="sessionRepository">The chat session repository.</param>
     /// <param name="messageRepository">The chat message repository.</param>
+    /// <param name="participantRepository">The chat participant repository.</param>
     /// <param name="sourceRepository">The chat memory resource repository.</param>
     /// <param name="promptsOptions">The prompts options.</param>
     public ChatHistoryController(
         ILogger<ChatHistoryController> logger,
         ChatSessionRepository sessionRepository,
         ChatMessageRepository messageRepository,
+        ChatParticipantRepository participantRepository,
         ChatMemorySourceRepository sourceRepository,
         IOptions<PromptsOptions> promptsOptions)
     {
         this._logger = logger;
         this._sessionRepository = sessionRepository;
         this._messageRepository = messageRepository;
+        this._participantRepository = participantRepository;
         this._sourceRepository = sourceRepository;
         this._promptOptions = promptsOptions.Value;
     }
 
     /// <summary>
     /// Create a new chat session and populate the session with the initial bot message.
+    /// The regex pattern that is used to match the user id will match the following format:
+    ///    - 2 period separated groups of one or more hyphen-delimited alphanumeric strings.
+    /// The pattern in meant to match two GUID-like strings in canonical textual representation
+    /// separated by a period (e.g., a1-2b-3c.x4-5y-6z)
     /// </summary>
-    /// <param name="chatParameters">Object that contains the parameters to create a new chat.</param>
+    /// <param name="userId">The user ID.</param>
+    /// <param name="chatParameter">Contains the title of the chat.</param>
     /// <returns>The HTTP action result.</returns>
     [HttpPost]
-    [Route("chatSession/create")]
+    [Route("chatSession/create/{userId:regex(([[a-z0-9]]+-)+[[a-z0-9]]+\\.([[a-z0-9]]+-)+[[a-z0-9]]+)}")]
```

Comment:
can we save this in a constant above and use it in both of the routes that have this regex?

Reply:
The regex is removed and replaced by a string from the body.